src/main/java/com/bootcamp/demo/service/UserService.java

```diff
@@ -0,0 +1,151 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class UserService implements IUserService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+    private static final String COLLECTION_USERS_PATH = "bookings/databases/users";
+    private final Firestore db;
+
+    public UserService(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public User createUser(final User user) {
+        LOGGER.info("CREATE USER - service function invoked: user = {}", user.toString());
+
+        try {
+            try {
+                this.getUserByEmail(user.getEmail());
+                throw new ServiceException("An account linked with this email already exists!");
+            } catch (ItemNotFoundException e) {
+                ApiFuture<WriteResult> collectionApiFuture = db.collection(COLLECTION_USERS_PATH)
+                        .document(user.getUserId())
+                        .set(user);
+
+                String updateTime = collectionApiFuture.get().getUpdateTime().toDate().toString();
+
+                LOGGER.info("CREATE USER - service function completed successfully. Update time: {}", updateTime);
+                return user;
+            }
+
+
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("CREATE USER - service function. Error message: {}", e.getMessage(), e);
+            throw new ServiceException(e);
+        }
+    }
+
+    @Override
+    public void deleteUserById(String userId) {
+        LOGGER.info("DELETE USER BY ID - service function invoked: user = {}", userId);
+
+        try {
+            DocumentSnapshot user = db.collection(COLLECTION_USERS_PATH)
+                    .document(userId)
+                    .get().get();
```

Comment:
You could either extract this to another method (to avoid its duplication here & in `getUserById`), or refactor `getUserById` to not throw an exception for missing item (and just return null) & use that method here.